src/pool/srv_target.c

```diff
@@ -472,13 +472,51 @@ pool_child_recreate(struct ds_pool_child *child)
 
 }
 
+struct ds_pool_flags {
+	struct ds_pool *pool;
+	uint32_t        disable_rebuild : 1, disable_dtx_resync : 1, immutable : 1;
+};
+
+static void
+apply_pool_flags(void *arg)
+{
+	struct ds_pool_flags *set_arg = arg;
+
+	if (set_arg->disable_rebuild)
+		set_arg->pool->sp_disable_rebuild = 1;
+	if (set_arg->disable_dtx_resync)
+		set_arg->pool->sp_disable_dtx_resync = 1;
+	if (set_arg->immutable)
+		set_arg->pool->sp_immutable = 1;
+}
+
+int
+ds_pool_apply_flags(struct ds_pool_flags *args)
+{
+	ABT_thread thread;
+	int        rc;
+
+	if (!args->disable_rebuild && !args->disable_dtx_resync && !args->immutable)
+		return 0;
+
+	rc = dss_ult_create(apply_pool_flags, args, DSS_XS_SYS, 0 /* tgt_idx */, 0 /* stack_size */,
+			    &thread);
```

Comment:
why need to create a ULT? cannot directly set it here?
if create ULT, why must on SYS XS0? cannot do it in self XS?